File: experiments/runs/2026-09-16_16-24-38_claim-office-example-mapping_exact-sol-v1.5-tcr-parity-domain-trial-cc_opus-5-no-thinking/src/price-list.ts
export interface Item {
  type: string;
  material?: string;
  enchantment?: number;
  cursed?: boolean;
}

interface PriceListEntry {
  insuranceValue: number;
  basePremium: number;
}

/** The MHPCO price list: it defines both what is insurable and at what value and premium. */
const PRICE_LIST: Record<string, PriceListEntry> = {
  sword: { insuranceValue: 1000, basePremium: 100 },
  amulet: { insuranceValue: 600, basePremium: 60 },
  staff: { insuranceValue: 800, basePremium: 80 },
  potion: { insuranceValue: 400, basePremium: 40 },
  rune: { insuranceValue: 250, basePremium: 25 },
  moonstone: { insuranceValue: 250, basePremium: 25 },
};

const COMPONENT_TYPES = ["rune", "moonstone"];

/** The MHPCO insures only the item types on its price list. */
function priceListEntry(type: string): PriceListEntry {
  const entry = PRICE_LIST[type];
  if (entry === undefined) {
    throw new Error(`MHPCO does not insure items of type "${type}"`);
  }
  return entry;
}

export function basePremiumOf(type: string): number {
  return priceListEntry(type).basePremium;
}

export function isComponent(item: Item): boolean {
  return COMPONENT_TYPES.includes(item.type);
}

/** The insurance sum is the unmodified insured value of every item on the policy. */
export function insuranceSum(items: Item[]): number {
  return items.reduce((sum, item) => sum + priceListEntry(item.type).insuranceValue, 0);
}

File: experiments/runs/2026-09-20_10-40-56_claim-office-java-example-mapping_exact-ptdd-v1.1-refactor-subagent-cc_opus-5-no-thinking/src/main/java/PriceList.java
import java.math.BigDecimal;
import java.util.Map;

/**
 * The MHPCO price list: every kind of item the MHPCO insures, and what it is
 * tariffed at.
 *
 * <p>A kind of item is tariffed at an insurance value — what the MHPCO
 * undertakes to cover it for — and a base premium — what it charges to cover
 * it. The two always travel together: a kind the MHPCO has taken on has both,
 * and a kind it has not taken on has neither, so they are listed as one entry
 * per kind rather than as two lists that could disagree.
 *
 * <p>The MHPCO tariffs main items individually and components as a category:
 * every component is listed at the same tariff, whatever it is made of.
 *
 * <p>Changes whenever the MHPCO revises its tariff or takes on a new kind of
 * item; knows nothing about how a premium is assembled from those amounts.
 */
final class PriceList {

    /**
     * What one kind of item is tariffed at.
     */
    private record Tariff(int insuranceValueInG, int basePremiumInG) {
    }

    private static final Tariff COMPONENT = new Tariff(250, 25);

    private static final Map<String, Tariff> TARIFFS = Map.of(
            "sword", new Tariff(1000, 100),
            "amulet", new Tariff(600, 60),
            "staff", new Tariff(800, 80),
            "potion", new Tariff(400, 40),
            "rune", COMPONENT,
            "moonstone", COMPONENT);

    private PriceList() {
    }

    static BigDecimal basePremiumFor(Item item) {
        return BigDecimal.valueOf(tariffFor(item).basePremiumInG());
    }

    static BigDecimal insuranceValueFor(Item item) {
        return BigDecimal.valueOf(tariffFor(item).insuranceValueInG());
    }

    /**
     * The MHPCO insures only what its price list names: an item of any other
     * kind is refused.
     */
    private static Tariff tariffFor(Item item) {
        Tariff tariff = TARIFFS.get(item.type());
        if (tariff == null) {
            throw new RejectedScenario("the MHPCO does not insure a " + item.type());
        }
        return tariff;
    }
}
